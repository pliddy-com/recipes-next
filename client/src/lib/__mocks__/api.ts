export interface ApiTestProps {
  getNavTaxonomy: () => [{ slug: string }];
  getRecipeSlugs: () => string[];
  getTagSlugs: () => string[];
  getRecipeIndex: () => [{ slug: string }];
  getRecipeList: () => [{ slug: string }];
  getRecipePage: () => { slug: string; title: string };
  getTagIndex: () => [{ slug: string; title: string }];
}

jest.mock('urql');

// import api library to mock
const api: ApiTestProps = jest.createMockFromModule('lib/api');

api.getNavTaxonomy = jest.fn().mockResolvedValue({
  categories: [
    { slug: 'slug-1', title: 'Title 1' },
    { slug: 'slug-2', title: 'Title 2' }
  ],
  cuisine: [
    { slug: 'slug-1', title: 'Title 1' },
    { slug: 'slug-2', title: 'Title 2' }
  ],
  tags: [
    { slug: 'slug-1', title: 'Title 1' },
    { slug: 'slug-2', title: 'Title 2' }
  ]
});

api.getRecipeSlugs = jest.fn().mockResolvedValue(['slug-1', 'slug-2']);

api.getTagSlugs = jest.fn().mockResolvedValue(['slug-1', 'slug-2']);

api.getRecipeIndex = jest.fn().mockResolvedValue([
  {
    slug: 'slug-1',
    title: 'Title 1'
  },
  {
    slug: 'slug-2',
    title: 'Title 2'
  }
]);

api.getRecipeList = jest.fn().mockResolvedValue([
  {
    __typename: 'Tag',
    slug: 'baking',
    title: 'Baking',
    linkedFrom: {
      recipeCollection: {
        total: 3,
        items: [
          {
            __typename: 'Recipe',
            sys: {
              firstPublishedAt: '2022-06-17T00:02:14.528Z',
              __typename: 'Sys'
            },
            title: 'Biscuits',
            slug: 'biscuits',
            abstract:
              'Traditional baking powder biscuits that use grated frozen butter and folded dough to give the biscuits a soft, flaky texture.',
            image: {
              __typename: 'Asset',
              title: 'Biscuits',
              description:
                'A batch of baking soda biscuits on a parchment-lined baking sheet.',
              contentType: 'image/jpeg',
              fileName: 'biscuits.jpg',
              size: 2583242,
              url: 'https://images.ctfassets.net/fo9qwg6zarbt/B6C4D23gXNDNfGVrxb1DK/9d8e62c1faf660f7e542eed14e044b57/biscuits.jpg',
              height: 3024,
              width: 4032
            },
            tagsCollection: {
              items: [
                {
                  __typename: 'Tag',
                  title: 'Baking',
                  slug: 'baking'
                },
                {
                  __typename: 'Tag',
                  title: 'Biscuits',
                  slug: 'biscuits'
                },
                {
                  __typename: 'Tag',
                  title: 'Breakfast',
                  slug: 'breakfast'
                },
                {
                  __typename: 'Tag',
                  title: 'Southern',
                  slug: 'southern'
                }
              ],
              __typename: 'RecipeTagsCollection'
            }
          },
          {
            __typename: 'Recipe',
            sys: {
              firstPublishedAt: '2022-11-15T21:50:41.051Z',
              __typename: 'Sys'
            },
            title: 'No-Knead Olive Bread',
            slug: 'no-knead-olive-bread',
            abstract:
              'A savory variation of no-knead bread with the addition of olives, garlic, and rosemary to give it a Mediterranean twist.',
            image: {
              __typename: 'Asset',
              title: 'Olive Bread',
              description: 'Loaf of no-knead olive bread',
              contentType: 'image/jpeg',
              fileName: 'olive-bread.JPG',
              size: 2849709,
              url: 'https://images.ctfassets.net/fo9qwg6zarbt/2rUa4IhfkLtRRBmHS6ihcD/e953337e00d1eba9551b15e7eee18229/olive-bread.JPG',
              height: 3024,
              width: 4032
            },
            tagsCollection: {
              items: [
                {
                  __typename: 'Tag',
                  title: 'Baking',
                  slug: 'baking'
                },
                {
                  __typename: 'Tag',
                  title: 'Bread',
                  slug: 'bread'
                },
                {
                  __typename: 'Tag',
                  title: 'Olives',
                  slug: 'olives'
                },
                {
                  __typename: 'Tag',
                  title: 'Mediterranean',
                  slug: 'mediterranean'
                }
              ],
              __typename: 'RecipeTagsCollection'
            }
          },
          {
            __typename: 'Recipe',
            sys: {
              firstPublishedAt: '2022-06-18T22:42:01.107Z',
              __typename: 'Sys'
            },
            title: 'No-Knead Bread',
            slug: 'no-knead-bread',
            abstract:
              'No-knead bread using instant yeast and a dutch oven or covered ceramic bread pan for a classic, crispy crust.',
            image: {
              __typename: 'Asset',
              title: 'No Knead Bread',
              description:
                'Loaf of no-knead bread cooling on a wire baking rack.',
              contentType: 'image/jpeg',
              fileName: 'no-knead-bread.JPG',
              size: 2943075,
              url: 'https://images.ctfassets.net/fo9qwg6zarbt/6HZDSm8AK4iQHr9v2UOGpq/17569d5be00441216338c7800e5ebc2c/no-knead-bread.JPG',
              height: 3024,
              width: 4032
            },
            tagsCollection: {
              items: [
                {
                  __typename: 'Tag',
                  title: 'Baking',
                  slug: 'baking'
                },
                {
                  __typename: 'Tag',
                  title: 'Bread',
                  slug: 'bread'
                }
              ],
              __typename: 'RecipeTagsCollection'
            }
          }
        ],
        __typename: 'RecipeCollection'
      },
      __typename: 'TagLinkingCollections'
    }
  }
]);

api.getRecipePage = jest.fn().mockResolvedValue({
  recipe: {
    title: 'Recipe Title',
    slug: 'slug-1',
    __typename: 'Recipe',
    sys: { firstPublishedAt: '2022-06-26T07:24:30.384Z' },
    abstract: 'abstract',
    recipeYield: 4,
    prepTime: 15,
    cookTime: 60,
    keywords: ['keyword1', 'keyword2'],
    image: {
      sys: {
        id: 'sysid-1',
        __typename: 'Sys'
      },
      __typename: 'Asset',
      title: 'Image Title',
      description: 'Image description.',
      contentType: 'image/jpeg',
      fileName: 'filename.jpg',
      size: 99999,
      url: 'https://recipes.pliddy.com/url',
      height: 300,
      width: 400
    },
    ingredientsList: [
      {
        sectionTitle: 'Section Label 1',
        sectionItems: ['ingredient 1', 'ingredient 2']
      }
    ],
    equipment: ['equipment 1', 'equipment 2'],
    instructionsList: [
      {
        sectionTitle: 'Section Label 1',
        sectionItems: ['Instructions item 1']
      },
      {
        sectionTitle: 'Section Label 2',
        sectionItems: ['Instructions item 2', 'Instructions item 3']
      }
    ],
    notes: ['note 1'],
    tagsCollection: {
      items: [
        {
          sys: {
            id: 'sysid-4',
            __typename: 'Sys'
          },
          __typename: 'Tag',
          title: 'Tag 1',
          slug: 'tag-1'
        }
      ],
      __typename: 'RecipeTagsCollection'
    }
  },
  cuisine: [
    {
      __typename: 'Tag',
      title: 'Tag 1 Title',
      slug: 'tag-1',
      linkedFrom: {
        recipeCollection: {
          total: 2,
          items: [
            {
              __typename: 'Recipe',
              slug: 'slug-1',
              title: 'Title 2'
            },
            {
              __typename: 'Recipe',
              slug: 'slug-2',
              title: 'Title 1'
            }
          ]
        }
      }
    },
    {
      __typename: 'Tag',
      title: 'Tag 2 Title',
      slug: 'tag-2',
      linkedFrom: {
        recipeCollection: {
          total: 2,
          items: [
            {
              __typename: 'Recipe',
              slug: 'slug-3',
              title: 'Title 3'
            },
            {
              __typename: 'Recipe',
              slug: 'slug-4',
              title: 'Title 5'
            }
          ]
        }
      }
    }
  ],
  categories: [
    {
      __typename: 'Tag',
      title: 'Tag 1 Title',
      slug: 'tag-1',
      linkedFrom: {
        recipeCollection: {
          total: 2,
          items: [
            {
              __typename: 'Recipe',
              slug: 'slug-1',
              title: 'Title 2'
            },
            {
              __typename: 'Recipe',
              slug: 'slug-2',
              title: 'Title 1'
            }
          ]
        }
      }
    },
    {
      __typename: 'Tag',
      title: 'Tag 2 Title',
      slug: 'tag-2',
      linkedFrom: {
        recipeCollection: {
          total: 2,
          items: [
            {
              __typename: 'Recipe',
              slug: 'slug-3',
              title: 'Title 3'
            },
            {
              __typename: 'Recipe',
              slug: 'slug-4',
              title: 'Title 5'
            }
          ]
        }
      }
    }
  ]
});

api.getTagIndex = jest.fn().mockResolvedValue([
  {
    __typename: 'Tag',
    title: 'Tag 1 Title',
    slug: 'tag-1',
    linkedFrom: {
      recipeCollection: {
        total: 2,
        items: [
          {
            __typename: 'Recipe',
            slug: 'slug-1',
            title: 'Title 2'
          },
          {
            __typename: 'Recipe',
            slug: 'slug-2',
            title: 'Title 1'
          }
        ]
      }
    }
  },
  {
    __typename: 'Tag',
    title: 'Tag 2 Title',
    slug: 'tag-2',
    linkedFrom: {
      recipeCollection: {
        total: 2,
        items: [
          {
            __typename: 'Recipe',
            slug: 'slug-3',
            title: 'Title 3'
          },
          {
            __typename: 'Recipe',
            slug: 'slug-4',
            title: 'Title 5'
          }
        ]
      }
    }
  }
]);

module.exports = api;
