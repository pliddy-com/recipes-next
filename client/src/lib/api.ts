/* istanbul ignore file */

import { queryGraphQLContent } from './gqlClient';
// import * as fs from 'fs';

import {
  filterSlugs,
  filterTagsWithRecipes,
  filterTaxonomyItemsWithRecipes
} from './apiFilters';

import {
  NavMenuDataDocument,
  RecipeIndexDocument,
  RecipeIndexQueryVariables,
  RecipeListDocument,
  RecipeListQueryVariables,
  RecipePageDocument,
  RecipePageQueryVariables,
  RecipeSlugsDocument,
  RecipeSlugsQueryVariables,
  TagSlugsDocument,
  TagSlugsQueryVariables,
  NavMenuDataQuery,
  RecipeSlugsQuery,
  TagSlugsQuery,
  RecipeListQuery,
  LinkedEntriesQuery,
  LinkedEntriesDocument
} from '../types/queries';

import { hasValue } from './utils';

// const ACCESS_TOKEN = process.env.NEXT_PUBLIC_CONTENTFUL_MANAGEMENT_TOKEN ?? '';
// const SPACE_ID = process.env.NEXT_PUBLIC_CONTENTFUL_SPACE_ID ?? '';
// const API_ENDPOINT = process.env.NEXT_PUBLIC_CONTENTFUL_CONTENT_API;

// const client = contentful.createClient({
//   accessToken: ACCESS_TOKEN,
//   space: SPACE_ID
// });

// used to query content for nav menu taxonomy on client layout
export const getNavTaxonomy = async () => {
  const { queryGraphQLContent } = await import('lib/gqlClient');

  const results: NavMenuDataQuery = await queryGraphQLContent(
    NavMenuDataDocument
  );

  const { categories, cuisine, tags } = results;

  return {
    ...(categories && {
      categories: filterTaxonomyItemsWithRecipes({
        taxonomyCollection: categories
      })
    }),
    ...(cuisine && {
      cuisine: filterTaxonomyItemsWithRecipes({ taxonomyCollection: cuisine })
    }),
    ...(tags && {
      tags: filterTagsWithRecipes({ tagCollection: tags })
    })
  };
};

// used to get linked entries for recipe edit

export const getLinkedEntries = async () => {
  const { queryGraphQLContent } = await import('lib/gqlClient');

  const results: LinkedEntriesQuery = await queryGraphQLContent(
    LinkedEntriesDocument
  );

  const { images, tags } = results;

  return {
    ...(images && {
      images: images.items
    }),
    ...(tags && {
      tags: tags.items
    })
  };
};

// used by getStaticProps for recipe page
export const getRecipeSlugs = async (variables: RecipeSlugsQueryVariables) => {
  const results: RecipeSlugsQuery = await queryGraphQLContent(
    RecipeSlugsDocument,
    variables
  );

  const { recipeCollection } = results;

  return recipeCollection && recipeCollection.items
    ? filterSlugs(recipeCollection.items)
    : [];
};

// used by getStaticProps for tag page
export const getTagSlugs = async (variables?: TagSlugsQueryVariables) => {
  const results: TagSlugsQuery = await queryGraphQLContent(
    TagSlugsDocument,
    variables
  );

  const { tagCollection } = results;

  const tags =
    tagCollection &&
    filterTagsWithRecipes({
      tagCollection
    });

  return tags ? filterSlugs(tags) : [];
};

// used to query content for home index page
export const getRecipeIndex = async (variables?: RecipeIndexQueryVariables) => {
  const { recipeCollection } = await queryGraphQLContent(
    RecipeIndexDocument,
    variables
  );

  return recipeCollection ? recipeCollection.items.filter(hasValue) : [];
};

// used to query content for tag & category page
export const getRecipeList = async (variables?: RecipeListQueryVariables) => {
  const { tagCollection } = await queryGraphQLContent(
    RecipeListDocument,
    variables
  );

  return tagCollection ? tagCollection.items.filter(hasValue) : [];
};

// used to query content for standalone recipe page
export const getRecipePage = async (variables?: RecipePageQueryVariables) => {
  const { recipeCollection, categoriesTaxonomy, cuisineTaxonomy } =
    await queryGraphQLContent(RecipePageDocument, variables);

  const categories =
    categoriesTaxonomy?.items.filter(hasValue)[0].childrenCollection?.items;

  const cuisine =
    cuisineTaxonomy?.items.filter(hasValue)[0].childrenCollection?.items;

  const recipe = recipeCollection?.items.filter(hasValue)[0];

  return {
    ...(recipe && { recipe }),
    ...(categories && { categories }),
    ...(cuisine && { cuisine })
  };
};

// used to query content for tag index page
export const getTagIndex = async (variables?: RecipeListQueryVariables) => {
  const results: RecipeListQuery = await queryGraphQLContent(
    RecipeListDocument,
    variables
  );

  const { tagCollection } = results;

  return tagCollection
    ? filterTagsWithRecipes({
        tagCollection
      })
    : [];
};
