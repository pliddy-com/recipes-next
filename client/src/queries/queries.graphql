# query for Taxonomy with Category & Subcategory levels
# getNavTaxonomy

query navMenuData {
  categories: taxonomyCollection(
    where: { slug: "categories" }
    order: title_ASC
    limit: 1
  ) {
    ...NavTaxonomy
  }

  cuisine: taxonomyCollection(
    where: { slug: "cuisine" }
    order: title_ASC
    limit: 1
  ) {
    ...NavTaxonomy
  }

  tags: tagCollection(order: title_ASC) {
    ...TagTaxonomy
  }
}

# Query for recipe slugs
# getRecipeSlugs

query recipeSlugs($where: RecipeFilter) {
  recipeCollection(where: $where, order: title_ASC) {
    __typename
    items {
      ...RecipeSlug
    }
  }
}

# query for category slugs
# getTagSlugs

query tagSlugs($where: TagFilter) {
  tagCollection(where: $where, order: title_ASC) {
    ...TagTaxonomy
  }
}

# query for lists of linked entries for tags and images
# getLinkedEntries

query linkedEntries {
  tags: tagCollection(order: title_ASC) {
    items {
      ...TagDefault
    }
  }
  images: assetCollection(order: fileName_ASC) {
    items {
      ...ImageDefault
    }
  }
}

# Query for List Page RecipeCollection (all recipes matching tag)
# getRecipeList

query recipeList($where: TagFilter) {
  __typename
  tagCollection(where: $where, order: title_ASC, limit: 1) {
    __typename
    total
    items {
      ...ListPageItem
    }
  }
}

# Query for Recipe Collection for home (index) page
# getRecipeIndex

query recipeIndex($where: RecipeFilter) {
  __typename
  recipeCollection(where: $where, order: title_ASC) {
    total
    items {
      ...RecipeSummary
    }
  }
}

# Query for Recipe page
# getRecipePage

query recipePage($where: RecipeFilter) {
  recipeCollection(where: $where, order: title_ASC, limit: 1) {
    __typename
    items {
      ...RecipeDefault
    }
  }

  categoriesTaxonomy: taxonomyCollection(
    where: { slug: "categories" }
    order: title_ASC
    limit: 1
  ) {
    ...SchemaTaxonomy
  }

  cuisineTaxonomy: taxonomyCollection(
    where: { slug: "cuisine" }
    order: title_ASC
    limit: 1
  ) {
    ...SchemaTaxonomy
  }
}

query recipeEntryQuery($id: String!) {
  recipe(id: $id) {
    sys {
      id
      publishedVersion
    }

    ingredientsList
    instructionsList
  }
}

# used for recipe content structure migration
query recipeCollectionQuery {
  recipeCollection {
    total
    items {
      sys {
        id
        publishedVersion
      }

      ingredientsList
      instructionsList
    }
  }
}
