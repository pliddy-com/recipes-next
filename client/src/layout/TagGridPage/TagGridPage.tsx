import { Suspense } from 'react';

import dynamic from 'next/dynamic';
// import { useRouter } from 'next/router';

import Box from '@mui/material/Box';
import Container from '@mui/material/Container';
import Grid from '@mui/material/Grid';
import Link from '@mui/material/Link';
import Typography from '@mui/material/Typography';

import ArrowRightIcon from '@mui/icons-material/ArrowRight';

import Loading from 'components/Loading/Loading';

import { ListPageItemFragment } from 'types/queries';

const PreviewCard = dynamic(
  () =>
    import(
      /* webpackChunkName: 'PreviewCard' */ 'components/PreviewCard/PreviewCard'
    ),
  { suspense: true }
);

interface TagGridPageProps {
  tags: (ListPageItemFragment | null)[];
}

const TagGridPage = ({ tags }: TagGridPageProps) => {
  // const { query } = useRouter();
  // const [scrollPage, setScrollPage] = useState<string | string[] | undefined>();

  return tags && tags.length > 0 ? (
    <Container className="page tagGrid" data-testid="page" maxWidth="xl">
      <Typography variant="h1">Tag Collections</Typography>
      <Typography variant="subtitle1" component="h2">
        {tags && `${tags.length} Tags`}
      </Typography>

      {tags.map((tag) => {
        const { slug, title, linkedFrom } = tag ?? {};

        const { recipeCollection } = linkedFrom ?? {};
        const { items } = recipeCollection ?? {};

        items &&
          items.sort((a, b) =>
            a && b && a.slug && b.slug && a.slug > b.slug ? 1 : -1
          );

        return slug && title && items ? (
          <Box key={slug}>
            <Link href={`/tags/${slug}`} underline="none" variant="h2">
              {title} <ArrowRightIcon />
            </Link>
            <Grid container spacing={2} mb={4}>
              {items &&
                items.map((recipe, index) => (
                  <Grid item lg={3} md={6} sm={12} xs={12} key={recipe?.slug}>
                    {recipe && (
                      <Suspense fallback={<Loading />}>
                        <PreviewCard recipe={recipe} preloadImg={index < 2} />
                      </Suspense>
                    )}
                  </Grid>
                ))}
            </Grid>
          </Box>
        ) : null;
      })}
    </Container>
  ) : null;
};

export default TagGridPage;
