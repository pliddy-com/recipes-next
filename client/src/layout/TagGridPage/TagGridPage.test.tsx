import '@testing-library/jest-dom';
import { act, render, waitFor } from '@testing-library/react';
import preloadAll from 'jest-next-dynamic';

import TagGridPage from './TagGridPage';
import { ListPageItemFragment } from 'types/queries';

import * as api from 'lib/api';

jest.mock('lib/api');

describe('TagGridPage', () => {
  afterEach(() => {
    jest.resetModules();
  });

  beforeAll(async () => {
    await preloadAll();
  });

  describe('when there is page content', () => {
    it('it renders the page', async () => {
      const tags = await api.getTagIndex({});
      const { asFragment, queryByTestId } = render(
        <TagGridPage tags={tags as (ListPageItemFragment | null)[]} />
      );

      // assert that content is rendered
      await act(async () =>
        waitFor(() => expect(queryByTestId('page')).toBeInTheDocument())
      );

      // assert that the component matches the existing snapshot
      expect(asFragment()).toMatchSnapshot();
    });
  });

  describe('when there is no pageContent', () => {
    it('it does not render the page', async () => {
      const tags = undefined;

      const { queryByTestId } = render(
        <TagGridPage tags={tags as unknown as ListPageItemFragment[]} />
      );

      await act(async () =>
        waitFor(() => expect(queryByTestId('page')).toBeNull())
      );
    });
  });

  describe('when there is missing tag information', () => {
    it('it does not render the content', () => {
      const tags = [null];

      render(<TagGridPage tags={tags} />);

      // test if card compoent is not rendered
      const card = document.querySelector('.MuiCard-root');
      expect(card).toBeNull();
    });
  });

  describe('when there is missing slug information', () => {
    it('it does not sort the content', async () => {
      const tags: ListPageItemFragment[] = [
        {
          __typename: 'Tag',
          title: 'Tag 1 Title',
          slug: 'tag-1',
          linkedFrom: {
            recipeCollection: {
              total: 2,
              items: [
                {
                  __typename: 'Recipe',
                  sys: { firstPublishedAt: '2022-06-26T07:24:30.384Z' },
                  slug: null,
                  title: 'Title 2'
                },
                {
                  __typename: 'Recipe',
                  sys: { firstPublishedAt: '2022-06-26T07:24:30.384Z' },
                  slug: 'title-1',
                  title: 'Title 1'
                }
              ]
            }
          }
        }
      ];

      const { asFragment } = render(<TagGridPage tags={tags} />);

      await act(async () =>
        waitFor(() => expect(asFragment()).toMatchSnapshot())
      );
    });
  });
});
