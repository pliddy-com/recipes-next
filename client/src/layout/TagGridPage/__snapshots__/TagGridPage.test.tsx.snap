// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TagGridPage when there is missing slug information it does not sort the content 1`] = `
<DocumentFragment>
  <div
    class="MuiContainer-root MuiContainer-maxWidthXl page tagGrid css-19r6kue-MuiContainer-root"
    data-testid="page"
  >
    <h1
      class="MuiTypography-root MuiTypography-h1 css-o2w69a-MuiTypography-root"
    >
      Tag Collections
    </h1>
    <h2
      class="MuiTypography-root MuiTypography-subtitle1 css-10wpov9-MuiTypography-root"
    >
      1 Tags
    </h2>
    <div
      class="MuiBox-root css-0"
    >
      <a
        class="MuiTypography-root MuiTypography-h2 MuiLink-root MuiLink-underlineNone css-2eo3m5-MuiTypography-root-MuiLink-root"
        href="/tags/tag-1"
      >
        Tag 1 Title 
        <svg
          aria-hidden="true"
          class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
          data-testid="ArrowRightIcon"
          focusable="false"
          viewBox="0 0 24 24"
        >
          <path
            d="m10 17 5-5-5-5v10z"
          />
        </svg>
      </a>
      <div
        class="MuiGrid-root MuiGrid-container MuiGrid-spacing-xs-2 css-71btut-MuiGrid-root"
      >
        <div
          class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 MuiGrid-grid-sm-12 MuiGrid-grid-md-6 MuiGrid-grid-lg-3 css-72mr71-MuiGrid-root"
        >
          <div
            class="MuiPaper-root MuiPaper-outlined MuiPaper-rounded MuiCard-root preview css-1v5z4dq-MuiPaper-root-MuiCard-root"
          >
            <a
              class="MuiButtonBase-root MuiCardActionArea-root css-b6lsxj-MuiButtonBase-root-MuiCardActionArea-root"
              href="/recipes/title-1"
              tabindex="0"
            >
              <div
                class="MuiCardContent-root css-46bh2p-MuiCardContent-root"
              >
                <div
                  class="MuiCardHeader-root css-185gdzj-MuiCardHeader-root"
                >
                  <div
                    class="MuiCardHeader-content css-1qbkelo-MuiCardHeader-content"
                  >
                    <span
                      class="MuiTypography-root MuiTypography-h5 MuiCardHeader-title css-1qvr50w-MuiTypography-root"
                    >
                      Title 1
                    </span>
                  </div>
                </div>
              </div>
              <span
                class="MuiCardActionArea-focusHighlight css-1v2exvi-MuiCardActionArea-focusHighlight"
              />
              <span
                class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
              />
            </a>
          </div>
        </div>
        <div
          class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 MuiGrid-grid-sm-12 MuiGrid-grid-md-6 MuiGrid-grid-lg-3 css-72mr71-MuiGrid-root"
        >
          <div
            class="MuiPaper-root MuiPaper-outlined MuiPaper-rounded MuiCard-root preview css-1v5z4dq-MuiPaper-root-MuiCard-root"
          >
            <a
              class="MuiButtonBase-root MuiCardActionArea-root css-b6lsxj-MuiButtonBase-root-MuiCardActionArea-root"
              href="/recipes/null"
              tabindex="0"
            >
              <div
                class="MuiCardContent-root css-46bh2p-MuiCardContent-root"
              >
                <div
                  class="MuiCardHeader-root css-185gdzj-MuiCardHeader-root"
                >
                  <div
                    class="MuiCardHeader-content css-1qbkelo-MuiCardHeader-content"
                  >
                    <span
                      class="MuiTypography-root MuiTypography-h5 MuiCardHeader-title css-1qvr50w-MuiTypography-root"
                    >
                      Title 2
                    </span>
                  </div>
                </div>
              </div>
              <span
                class="MuiCardActionArea-focusHighlight css-1v2exvi-MuiCardActionArea-focusHighlight"
              />
              <span
                class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
              />
            </a>
          </div>
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;

exports[`TagGridPage when there is page content it renders the page 1`] = `
<DocumentFragment>
  <div
    class="MuiContainer-root MuiContainer-maxWidthXl page tagGrid css-19r6kue-MuiContainer-root"
    data-testid="page"
  >
    <h1
      class="MuiTypography-root MuiTypography-h1 css-o2w69a-MuiTypography-root"
    >
      Tag Collections
    </h1>
    <h2
      class="MuiTypography-root MuiTypography-subtitle1 css-10wpov9-MuiTypography-root"
    >
      2 Tags
    </h2>
    <div
      class="MuiBox-root css-0"
    >
      <a
        class="MuiTypography-root MuiTypography-h2 MuiLink-root MuiLink-underlineNone css-2eo3m5-MuiTypography-root-MuiLink-root"
        href="/tags/tag-1"
      >
        Tag 1 Title 
        <svg
          aria-hidden="true"
          class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
          data-testid="ArrowRightIcon"
          focusable="false"
          viewBox="0 0 24 24"
        >
          <path
            d="m10 17 5-5-5-5v10z"
          />
        </svg>
      </a>
      <div
        class="MuiGrid-root MuiGrid-container MuiGrid-spacing-xs-2 css-71btut-MuiGrid-root"
      >
        <div
          class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 MuiGrid-grid-sm-12 MuiGrid-grid-md-6 MuiGrid-grid-lg-3 css-72mr71-MuiGrid-root"
        >
          <div
            class="MuiPaper-root MuiPaper-outlined MuiPaper-rounded MuiCard-root preview css-1v5z4dq-MuiPaper-root-MuiCard-root"
          >
            <a
              class="MuiButtonBase-root MuiCardActionArea-root css-b6lsxj-MuiButtonBase-root-MuiCardActionArea-root"
              href="/recipes/slug-1"
              tabindex="0"
            >
              <div
                class="MuiCardContent-root css-46bh2p-MuiCardContent-root"
              >
                <div
                  class="MuiCardHeader-root css-185gdzj-MuiCardHeader-root"
                >
                  <div
                    class="MuiCardHeader-content css-1qbkelo-MuiCardHeader-content"
                  >
                    <span
                      class="MuiTypography-root MuiTypography-h5 MuiCardHeader-title css-1qvr50w-MuiTypography-root"
                    >
                      Title 2
                    </span>
                  </div>
                </div>
              </div>
              <span
                class="MuiCardActionArea-focusHighlight css-1v2exvi-MuiCardActionArea-focusHighlight"
              />
              <span
                class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
              />
            </a>
          </div>
        </div>
        <div
          class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 MuiGrid-grid-sm-12 MuiGrid-grid-md-6 MuiGrid-grid-lg-3 css-72mr71-MuiGrid-root"
        >
          <div
            class="MuiPaper-root MuiPaper-outlined MuiPaper-rounded MuiCard-root preview css-1v5z4dq-MuiPaper-root-MuiCard-root"
          >
            <a
              class="MuiButtonBase-root MuiCardActionArea-root css-b6lsxj-MuiButtonBase-root-MuiCardActionArea-root"
              href="/recipes/slug-2"
              tabindex="0"
            >
              <div
                class="MuiCardContent-root css-46bh2p-MuiCardContent-root"
              >
                <div
                  class="MuiCardHeader-root css-185gdzj-MuiCardHeader-root"
                >
                  <div
                    class="MuiCardHeader-content css-1qbkelo-MuiCardHeader-content"
                  >
                    <span
                      class="MuiTypography-root MuiTypography-h5 MuiCardHeader-title css-1qvr50w-MuiTypography-root"
                    >
                      Title 1
                    </span>
                  </div>
                </div>
              </div>
              <span
                class="MuiCardActionArea-focusHighlight css-1v2exvi-MuiCardActionArea-focusHighlight"
              />
              <span
                class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
              />
            </a>
          </div>
        </div>
      </div>
    </div>
    <div
      class="MuiBox-root css-0"
    >
      <a
        class="MuiTypography-root MuiTypography-h2 MuiLink-root MuiLink-underlineNone css-2eo3m5-MuiTypography-root-MuiLink-root"
        href="/tags/tag-2"
      >
        Tag 2 Title 
        <svg
          aria-hidden="true"
          class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
          data-testid="ArrowRightIcon"
          focusable="false"
          viewBox="0 0 24 24"
        >
          <path
            d="m10 17 5-5-5-5v10z"
          />
        </svg>
      </a>
      <div
        class="MuiGrid-root MuiGrid-container MuiGrid-spacing-xs-2 css-71btut-MuiGrid-root"
      >
        <div
          class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 MuiGrid-grid-sm-12 MuiGrid-grid-md-6 MuiGrid-grid-lg-3 css-72mr71-MuiGrid-root"
        >
          <div
            class="MuiPaper-root MuiPaper-outlined MuiPaper-rounded MuiCard-root preview css-1v5z4dq-MuiPaper-root-MuiCard-root"
          >
            <a
              class="MuiButtonBase-root MuiCardActionArea-root css-b6lsxj-MuiButtonBase-root-MuiCardActionArea-root"
              href="/recipes/slug-3"
              tabindex="0"
            >
              <div
                class="MuiCardContent-root css-46bh2p-MuiCardContent-root"
              >
                <div
                  class="MuiCardHeader-root css-185gdzj-MuiCardHeader-root"
                >
                  <div
                    class="MuiCardHeader-content css-1qbkelo-MuiCardHeader-content"
                  >
                    <span
                      class="MuiTypography-root MuiTypography-h5 MuiCardHeader-title css-1qvr50w-MuiTypography-root"
                    >
                      Title 3
                    </span>
                  </div>
                </div>
              </div>
              <span
                class="MuiCardActionArea-focusHighlight css-1v2exvi-MuiCardActionArea-focusHighlight"
              />
              <span
                class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
              />
            </a>
          </div>
        </div>
        <div
          class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 MuiGrid-grid-sm-12 MuiGrid-grid-md-6 MuiGrid-grid-lg-3 css-72mr71-MuiGrid-root"
        >
          <div
            class="MuiPaper-root MuiPaper-outlined MuiPaper-rounded MuiCard-root preview css-1v5z4dq-MuiPaper-root-MuiCard-root"
          >
            <a
              class="MuiButtonBase-root MuiCardActionArea-root css-b6lsxj-MuiButtonBase-root-MuiCardActionArea-root"
              href="/recipes/slug-4"
              tabindex="0"
            >
              <div
                class="MuiCardContent-root css-46bh2p-MuiCardContent-root"
              >
                <div
                  class="MuiCardHeader-root css-185gdzj-MuiCardHeader-root"
                >
                  <div
                    class="MuiCardHeader-content css-1qbkelo-MuiCardHeader-content"
                  >
                    <span
                      class="MuiTypography-root MuiTypography-h5 MuiCardHeader-title css-1qvr50w-MuiTypography-root"
                    >
                      Title 5
                    </span>
                  </div>
                </div>
              </div>
              <span
                class="MuiCardActionArea-focusHighlight css-1v2exvi-MuiCardActionArea-focusHighlight"
              />
              <span
                class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
              />
            </a>
          </div>
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
